src/plugins/kibana_react/public/page/page_header/page_header.tsx

```diff
@@ -0,0 +1,142 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { FC } from 'react';
+import {
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIcon,
+  EuiTitle,
+  IconType,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { CoreStart } from 'kibana/public';
+import { RedirectAppLinks } from '../../app_links';
+import { useKibana } from '../../context';
+
+import './index.scss';
+
+interface Props {
+  hideToolbar?: boolean;
+  iconType?: IconType;
+  overlap?: boolean;
+  showDevToolsLink?: boolean;
+  showManagementLink?: boolean;
+  title: JSX.Element | string;
+}
+
+export const PageHeader: FC<Props> = ({
+  hideToolbar,
+  iconType,
+  overlap,
+  showDevToolsLink,
+  showManagementLink,
+  title,
+}) => {
+  const {
+    services: { application },
+  } = useKibana<CoreStart>();
+
+  const {
+    management: isManagementEnabled,
+    dev_tools: isDevToolsEnabled,
+  } = application.capabilities.navLinks;
+
+  return (
+    <header
+      className={`pageHeader ${overlap ? 'pageHeader--hasOverlap' : 'pageHeader--noOverlap'}`}
```

Comment:
Again all these classes ideally should be more specific and use the `kbn` prefix.